hazelcast/src/main/java/com/hazelcast/config/QueueConfig.java

```diff
@@ -207,7 +217,7 @@ public QueueStoreConfig getQueueStoreConfig() {
      * @param queueStoreConfig set the QueueStore configuration to this configuration
      * @return the QueueStore configuration
      */
-    public QueueConfig setQueueStoreConfig(QueueStoreConfig queueStoreConfig) {
+    public QueueConfig setQueueStoreConfig(@Nullable QueueStoreConfig queueStoreConfig) {
```

Comment:
As agreed, added warning on queue container startup because of two reasons:
- there is a limit set by default and the only way to disable it is to set it to `Integer.MAX_VALUE`
- I prefer not to do validation while the config is still being created, maybe the user is in the process of making it valid.